libcxx/test/libcxx/diagnostics/ranges.nodiscard_extensions.verify.cpp

```diff
@@ -87,4 +89,15 @@ void test() {
   std::ranges::unique(iter, iter); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
   std::ranges::upper_bound(range, 1); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
   std::ranges::upper_bound(iter, iter, 1); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+
+#if TEST_STD_VER >= 23
+  std::ranges::fold_left(range, 0, std::plus());
+  // expected-warning@-1{{ignoring return value of function declared with 'nodiscard' attribute}}
```

Comment:
I don't think this is correct. There are many valid ways to use this function without inspecting the result.

Reply:
Can you provide an example please? I'm not aware of any.

Reply:
If you agree that it's not good code, I don't really see the problem with warning here. There's lots of correct code that the compiler generates warnings on. You wouldn't want to remove `[[nodiscard]]` from `std::count` just because there are valid ways to use the function without inspecting the return value, would you?

Reply:
I think it's substantially better to be explicit about wanting to discard the result, since the primary function of this algorithm is to return a computed result. I also expect the common case to be to want the result, rather than not.

```cpp
(void)std::ranges::fold_left(r, 0, side_effect_f); // clearly discarding the result
```